MekHQ/src/mekhq/campaign/mission/AtBDynamicScenarioFactory.java

```diff
@@ -2419,77 +2568,357 @@ private static void correctNonAeroFlyerBehavior(List<Entity> entityList, int boa
+                    Arrays.fill(generatedBombs, BombType.B_NONE);
+                    generatedBombs[BombType.B_TAG]++;
+                    tagCount--;
+                    ((IBomber) curBomber).setBombChoices(generatedBombs);
                 }
 
-                loadBombs(entity, validBombChoices, campaign.getGameYear());
-                numBombers++;
             }
+
         }
+
     }
 
     /**
-     * Worker function that takes an entity and an array of bomb types
-     * and loads it up with as many of a mostly period-appropriate random bomb type
-     * as it's capable of holding
+     * Randomly generate a set of external ordnance up to the number of indicated bomb units. Lower
+     * rated forces are more likely to get simpler types (HE and rockets).
+     * @param bombUnits   how many bomb units to generate, some types count as more than one unit so
+     *                    returned counts may be lower than this but never higher
+     * @param airOnly     true to only select air-to-air ordnance
+     * @param isPirate    true if force is pirate, specific low-tech/high chaos selections
+     * @param quality     force rating to work with
+     * @param year        current year, for tech filter
+     * @return            array of integers, with each element being a bomb count using BombUnit
+     *                    enums as the lookup e.g. [BombUnit.HE] will get the number of HE
+     *                    bombs.
      */
-    private static void loadBombs(Entity entity, int[] validBombChoices, int year) {
-        int[] bombChoices = new int[BombType.B_NUM];
+    private static int[] generateExternalOrdnance (int bombUnits,
+                                                     boolean airOnly,
+                                                     boolean isPirate,
+                                                     int quality,
+                                                     int year) {
+
+        int[] bombLoad = new int[BombType.B_NUM];
+
+        if (bombUnits <= 0) {
+            return bombLoad;
+        }
+
+        // Get a random predefined loadout
+        String mapName = "";
+        double countWeight = 0.0;
+
+        Map<String, Integer> loadoutMap;
+        if (!isPirate) {
+            loadoutMap = bombMapGroundSpread;
+        } else {
+            loadoutMap = bombMapPirateGroundSpread;
+        }
+
+        double completeWeight = loadoutMap.values().stream().mapToDouble(curWeight -> Math.max(curWeight, 1.0)).sum();
+        double randomThreshold = (Compute.randomInt(100) / 100.0) * completeWeight;
+        for (String curMap : loadoutMap.keySet()) {
+            countWeight += Math.max(loadoutMap.get(curMap), 1.0);
+            if (countWeight >= randomThreshold) {
+                mapName = curMap;
+                break;
+            }
+        }
+
+        Map<Integer,Integer> bombMap;
+        if (!isPirate) {
+            if (!airOnly) {
+                switch (mapName) {
```

Comment:
This can all be significantly simplified.

Reply:
Not certain how. The test cannot be combined, as it needs to split between pirate and non-pirate loads, then it also needs to choose between air/ground and air/air loads - didn't feel that with only two choices of air/air loads, and the limited ordnance choices warranted an extra set of maps.

Granted, I could remove the negation, but the coding style is to put more likely choices at the top so when stepping through or manually reading code, large jumps to following lines are outliers.